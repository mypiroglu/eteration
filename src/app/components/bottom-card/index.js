import React from 'react';
import {View} from 'react-native';
import {Text, Button} from '../';
import styles from './style';

export const BottomCard = ({
  title = 'Add to cart',
  price = 0,
  onPress,
  style,
}) => {
  return (
    <View style={[styles.bottomContainer, style]}>
      <View>
        <Text style={styles.priceText}>Price:</Text>
        <Text style={styles.priceNumber}>{price} â‚º</Text>
      </View>
      <Button title={title} onPress={onPress} />
    </View>
  );
};
