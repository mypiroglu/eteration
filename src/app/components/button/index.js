import React from 'react';
import {TouchableOpacity} from 'react-native';
import {Text} from '..';
import styles from './style';
export const Button = ({
  title,
  onPress,
  style,
  disabled = false,
  textStyle,
}) => {
  return (
    <TouchableOpacity
      style={
        disabled
          ? [styles.button, style, {backgroundColor: '#000'}]
          : [styles.button, style]
      }
      onPress={onPress}
      disabled={disabled}>
      <Text style={[styles.text, textStyle]}>{title}</Text>
    </TouchableOpacity>
  );
};
