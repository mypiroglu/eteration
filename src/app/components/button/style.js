import {StyleSheet} from 'react-native';
import colors from '../../utils/colors';

export default StyleSheet.create({
  button: {
    backgroundColor: colors.blue, // Özelleştirilebilir
    padding: 10,
    borderRadius: 5,
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    color: colors.white, // Özelleştirilebilir
    textAlign: 'center',
    fontWeight: '500',
  },
});
