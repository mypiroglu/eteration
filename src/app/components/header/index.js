import React from 'react';
import {Pressable, View} from 'react-native';
import {Icon, Text} from '..';
import styles from './style';

const HeaderIcon = ({icon, style, onPress}) => (
  <Pressable style={style} onPress={onPress}>
    <Icon icon={icon} />
  </Pressable>
);

export const Header = ({
  title,
  leftIcon,
  rightIcon,
  onLeftPress,
  onRightPress,
  style,
}) => {
  return (
    <View style={[styles.header, style]}>
      <Text style={styles.titleText}>{title}</Text>
      {leftIcon && (
        <HeaderIcon
          icon={leftIcon}
          style={styles.leftButton}
          onPress={onLeftPress}
        />
      )}
      {rightIcon && (
        <HeaderIcon
          icon={rightIcon}
          style={styles.rightButton}
          onPress={onRightPress}
        />
      )}
    </View>
  );
};
