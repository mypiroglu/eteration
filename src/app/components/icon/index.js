import React from 'react';
import {
  Basket,
  Home,
  Star,
  UnStar,
  User,
  Search,
  EyeOff,
  Eye,
} from '../../../assets/svg';

const iconTypes = {
  Basket,
  Home,
  Star,
  User,
  Search,
  EyeOff,
  Eye,
  UnStar,
};

export const Icon = ({icon, ...props}) => {
  let Icon = iconTypes[icon];
  return <Icon {...props} />;
};
