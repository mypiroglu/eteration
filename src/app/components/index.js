export * from './text-input';
export * from './button';
export * from './icon';
export * from './header';
export * from './text';
export * from './product-card';
export * from './info-product-card';
export * from './bottom-card';
export * from './radio-button';
