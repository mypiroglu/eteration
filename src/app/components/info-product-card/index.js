import React, {useState} from 'react';
import {View, Image, Pressable} from 'react-native';
import {Text, Button} from '..';
import styles from './style';
import {useNavigation} from '@react-navigation/native';
import {useDispatch} from 'react-redux';
import {plusOne, minusOne} from '../../redux/slice/basket-slice';
import {addToBasket} from '../../redux/slice/basket-slice';

export const InfoProductCard = ({data, preset = 'primary'}) => {
  const {item, quantity} = data;
  let {brand, price, image, description} = item;

  const dispatch = useDispatch();
  const [count, setCount] = useState(1);
  const navigation = useNavigation();
  const onHandlePress = () => {
    navigation.navigate('product-detail-screen', {item: item});
  };

  const onHandlePressAddToBasket = () => {
    // Burada sepete ekleme işlemi yapılacak. Sonra redux'a taşınacak. ASYNC STORAGE KULLANILACAK.
    console.log('button31');
    const quantity = 1;
    dispatch(addToBasket({item: item, quantity: quantity}));
  };
  const onHandlePressIncrease = () => {
    setCount(count + 1);
    dispatch(plusOne(item));
    // Burada sepete ekleme işlemi yapılacak. Sonra redux'a taşınacak. ASYNC STORAGE KULLANILACAK.
  };
  const onHandlePressDecrease = () => {
    if (count > 0) {
      setCount(count - 1);
      dispatch(minusOne(item));
    }
    // Burada sepete ekleme işlemi yapılacak. Sonra redux'a taşınacak. ASYNC STORAGE KULLANILACAK.
  };
  return (
    <Pressable style={styles.container} onPress={onHandlePress}>
      <Pressable style={styles.imageContainer} onPress={onHandlePress}>
        <Image source={{uri: image}} style={styles.image} />
      </Pressable>
      <View style={styles.infoContainer}>
        <View style={styles.info}>
          <Text numberOfLines={1} style={styles.brand}>
            {brand}
          </Text>
          <Text style={styles.price}>{price} TL</Text>
        </View>
        <View style={styles.info}>
          <Text numberOfLines={1} style={styles.description}>
            {description}
          </Text>
        </View>
        {preset === 'primary' ? (
          <View style={styles.infoButton}>
            <Button
              onPress={onHandlePressDecrease}
              title="-"
              style={styles.button}
            />
            <Text>{quantity}</Text>
            <Button
              onPress={onHandlePressIncrease}
              title="+"
              style={styles.button}
            />
          </View>
        ) : (
          <Button
            title="Add to Basket"
            onPress={onHandlePressAddToBasket}
            style={styles.buttonAdd}
          />
        )}
      </View>
    </Pressable>
  );
};
