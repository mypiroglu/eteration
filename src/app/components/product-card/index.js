import React, {useState, useEffect} from 'react';
import {View, Image, Pressable} from 'react-native';
import {Text, Button, Icon} from '..';
import styles from './style';
import {useNavigation} from '@react-navigation/native';
import {
  addToFavorites,
  removeFromFavorites,
} from '../../redux/slice/favorites-slice';
import {addToBasket, removeFromBasket} from '../../redux/slice/basket-slice';
import {useDispatch, useSelector} from 'react-redux';

export const ProductCard = ({item}) => {
  const dispatch = useDispatch();
  const {price, brand, image, description} = item;
  const [isFavorite, setIsFavorite] = useState(false);
  const navigation = useNavigation();
  const {favorites} = useSelector(state => state.favorites);
  const onHandlePress = () => {
    navigation.navigate('product-detail-screen', {item: item});
  };

  const onHandlePressIcon = () => {
    isFavorite
      ? dispatch(removeFromFavorites(item))
      : dispatch(addToFavorites(item));
    setIsFavorite(!isFavorite);
    // Burada favoriye ekleme işlemi yapılacak. Sonra redux'a taşınacak. ASYNC STORAGE KULLANILACAK.
  };
  const onHandlePressButton = () => {
    console.log('button');
    const quantity = 1;
    dispatch(addToBasket({item: item, quantity}));
    // Burada sepete ekleme işlemi yapılacak. Sonra redux'a taşınacak. ASYNC STORAGE KULLANILACAK.
  };
  useEffect(() => {
    const favoriteState = favorites.find(favorite => favorite.id === item.id);
    if (favoriteState) {
      setIsFavorite(true);
    } else {
      setIsFavorite(false);
    }
  }, [item]);
  return (
    <Pressable style={styles.root} onPress={onHandlePress}>
      <>
        <View style={styles.image}>
          <Pressable style={styles.icon} onPress={onHandlePressIcon}>
            <Icon
              icon={isFavorite ? 'Star' : 'UnStar'}
              width="30"
              height="30"
              style={styles.icon}
            />
          </Pressable>
          <Image
            source={{
              uri: image
                ? image
                : 'https://yekpar.com/writable/uploads/medias/files/istockphoto-1357365823-612x612.jpg',
            }}
            style={styles.image}
          />
          <Text style={styles.priceText}>
            {price ? price + ' ₺' : 'no price information'}
          </Text>
          <Text numberOfLines={1} style={styles.productText}>
            {brand ? brand : 'no title information'}
          </Text>
          <Text numberOfLines={1} style={styles.productText}>
            {description ? description : 'no description information'}
          </Text>
        </View>
        <Button
          title="Add to Cart"
          textStyle={styles.buttonText}
          style={styles.button}
          onPress={onHandlePressButton}
        />
      </>
    </Pressable>
  );
};
