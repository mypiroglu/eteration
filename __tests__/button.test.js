import React from 'react';
import {render} from '@testing-library/react-native';
import {Button} from '../src/app/components/button';

describe('<Button />', () => {
  it('matches the snapshot', () => {
    const {toJSON} = render(
      <Button title="Click Me" style={{backgroundColor: 'red'}} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
