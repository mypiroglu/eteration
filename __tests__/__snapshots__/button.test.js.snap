// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches the snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "red",
      "borderRadius": 5,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      [
        {
          "color": "#FFFFFF",
          "fontWeight": "500",
          "textAlign": "center",
        },
        undefined,
      ]
    }
    testID="text"
  >
    Click Me
  </Text>
</View>
`;
